test/integration/swim-test.js

```diff
@@ -0,0 +1,198 @@
+    var badStatuses = res.pingReqErrs.filter(function filterErr(err) {
+        return err.type === 'ringpop.ping-req.bad-ping-status';
+    });
+
+    assert.equals(badStatuses.length, num, 'correct number of bad statuses');
+}
+
+function mkNoGossip(cluster) {
+    var noop = function noop() {
+    };
+
+    cluster.forEach(function eachRingpop(ringpop) {
+        ringpop.gossip.start = noop;
+    });
+}
+
+function mkBadPingReqResponder(ringpop) {
+    ringpop.channel.register('/protocol/ping-req', function protocolPingReq(arg1, arg2, hostInfo, cb) {
+        cb(null, null, JSON.stringify('badbody'));
+    });
+}
+
+testRingpopCluster({
+    tap: function tap(cluster) {
+        mkNoGossip(cluster);
+    }
+}, 'ping-reqs 1 member', function t(bootRes, cluster, assert) {
+    var ringpop = cluster[0];
+    var unreachableMember = ringpop.membership.findMemberByAddress(cluster[1].hostPort);
+
+    sendPingReq({
+        ringpop: cluster[0],
+        unreachableMember: unreachableMember,
+        pingReqSize: 3
+    }, function onPingReq(err, res) {
+        assert.ifErr(err, 'no error occurred');
+        assert.equal(res.pingReqAddrs.length, 1,
+            'number of selected ping-req members is correct');
+        assert.ok(res.pingReqSuccess.address === cluster[2].hostPort,
+            'successful ping-req response from either member');
+        assert.end();
+    });
+});
+
+testRingpopCluster({
+    size: 5,
+    tap: function tap(cluster) {
+        mkNoGossip(cluster);
+    }
+}, 'ping-reqs pingReqSize members', function t(bootRes, cluster, assert) {
+    var ringpop = cluster[0];
+    var unreachableMember = ringpop.membership.
+        findMemberByAddress(cluster[1].hostPort);
+    var pingReqSize = 3;
+
+    sendPingReq({
+        ringpop: cluster[0],
+        unreachableMember: unreachableMember,
+        pingReqSize: pingReqSize
+    }, function onPingReq(err, res) {
+        assert.ifErr(err, 'no error occurred');
+        assert.equal(res.pingReqAddrs.length, pingReqSize,
+            'number of selected ping-req members is correct');
+        assert.equal(unreachableMember.status, 'alive',
+            'unreachable member is alive');
+        assert.end();
+    });
+});
+
+testRingpopCluster({
+    size: 5,
+    tap: function tap(cluster) {
+        mkNoGossip(cluster);
+    }
+}, 'ping-req target unreachable', function t(bootRes, cluster, assert) {
+    var badRingpop = cluster[4];
+    badRingpop.destroy();
+
+    var ringpop = cluster[0];
+    var unreachableMember = ringpop.membership.findMemberByAddress(badRingpop.hostPort);
+    var pingReqSize = 3;
+
+    sendPingReq({
+        ringpop: ringpop,
+        unreachableMember: unreachableMember,
+        pingReqSize: pingReqSize
+    }, function onPingReq(err, res) {
+        assert.ifErr(err, 'no error occurred');
+        assertNumBadStatuses(assert, res, pingReqSize);
+        assert.equals(unreachableMember.status, 'suspect',
```

Comment:
note: It seems fragile that `unreachableMember` is a mutated object. 

This test would seem more stable if you fetched the member from `ringpop.membership` again.